ceph/rados/crushtool_workflows.py

```diff
@@ -0,0 +1,649 @@
+            Returns a tuple consisting of the execution status and the bin file location
+            (True, {})
+        """
+        if not self.check_file_exists_on_client(loc=loc):
+            log.error(f"Bin file : {loc} not present on the Client")
+            return False, ""
+
+        # Setting the contents of the bin file in json
+        cmd = f"crushtool -i {loc} --dump -f json"
+        try:
+            out, err = self.client.exec_command(cmd=cmd, sudo=True)
+            crush_contents = json.loads(out)
+        except Exception as error:
+            log.error(f"Failed to get the contents of the bin file. error: {error}")
+            return False, ""
+        log.debug(
+            f"Successfully fetched the contents of the bin file : {crush_contents}"
+        )
+        return True, crush_contents
+
+    def test_crush_map_bin(
+        self, loc="/tmp/crush_modified.map.bin", test="show-statistics"
+    ) -> (bool, dict):
+        """Method to perform tests on the crush map provided
+        Args::
+            loc: Name of the source bin file that should be modified
+            test: Test to be performed on the crush map.
+              Allowed tests:
+                1. [ Default ] show-statistics - Displays a summary of the distribution.
+                2. show-mappings - Displays the mapping of each value in the range [--min-x,--max-x]
+                3. show-bad-mappings - Displays which value failed to be mapped to the required number of devices.
+                4. show-utilization - Displays the expected and actual utilization for each device,
+                                      for each number of replicas.
+                5. show-utilization-all - Displays the same as –show-utilization but does not suppress
+                                          output when the weight of a device is zero.
+                6. show-choose-tries - Displays how many attempts were needed to find a device mapping.
+
+        Examples::
+            test_crush_map_bin(
+                loc="/tmp/crush_modified.map.bin"
+                test="show-statistics"
+            )
+
+        Returns::
+        Tuple containing the execution status and the output of the test executed
+        (True, "bad mapping rule 1 x 781 num_rep 7 result [8,10,2,11,6,9]")
+        """
+        if not self.check_file_exists_on_client(loc=loc):
+            log.error(f"Bin file : {loc} not present on the Client")
+            return False, " "
+
+        # Setting the contents of the bin file in json
+        cmd = f"crushtool -i {loc} --test --{test}"
+        try:
+            out, err = self.client.exec_command(cmd=cmd, sudo=True)
+        except Exception as error:
+            log.error(f"Failed to test bin file with test : {test}. error: {error}")
+            return False, " "
+        log.debug(f"Successfully tested the bin file for {test} and output : {out}")
+        return True, out
+
+    def check_file_exists_on_client(self, loc) -> bool:
```

Comment:
Can we move this to the core_worflows.py where all common method exists.

Reply:
Yes, Will take care of this in next PR.